src/App.tsx

```diff
@@ -1,20 +1,29 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback } from 'react';
+import AddTodoForm from './components/AddTodoForm';
+import TodoList from './components/TodoList';
+import Footer from './components/Footer';
+import { useDispatchContext, useTodoContext } from './context/GlobalContext';
 
 export const App: React.FC = () => {
+  const { todos } = useTodoContext();
+  const { dispatch } = useDispatchContext();
+  const allActive = todos.filter(todo => !todo.completed).length > 0;
+
+  const toggleAll = useCallback(() => {
```

Comment:
Not a fan of trying to memoize the function here. 
allActive, todos => are arrays, react uses shallow compare for dependency array. 

From my perspective, it simply makes no sense and caching is more complex operation than the actual calculation here.